msal/src/main/java/com/microsoft/identity/client/internal/controllers/LocalMSALController.java

```diff
@@ -463,14 +468,224 @@ public boolean removeCurrentAccount(RemoveAccountCommandParameters parameters) t
+        // Logging start of method
+        final String methodName = ":deviceCodeFlowAuthRequest";
+        Logger.verbose(
+                TAG + methodName,
+                "Device Code Flow: Authorizing user code..."
+        );
+
+        // Default scopes here
+        final Set<String> mergedScopes = addDefaultScopes(parameters);
+
+        final DeviceCodeFlowCommandParameters parametersWithScopes = parameters
+                .toBuilder()
+                .scopes(mergedScopes)
+                .build();
+
+        logParameters(TAG, parametersWithScopes);
+
+        // Start telemetry with LOCAL_DEVICE_CODE_FLOW_ACQUIRE_URL_AND_CODE
+        Telemetry.emit(
+                new ApiStartEvent()
+                        .putProperties(parametersWithScopes)
+                        .putApiId(TelemetryEventStrings.Api.LOCAL_DEVICE_CODE_FLOW_ACQUIRE_URL_AND_CODE)
+        );
+
+        // Create OAuth2Strategy using commandParameters and strategyParameters
+        final OAuth2StrategyParameters strategyParameters = new OAuth2StrategyParameters();
+        strategyParameters.setContext(parametersWithScopes.getAndroidApplicationContext());
+
+        final OAuth2Strategy oAuth2Strategy = parametersWithScopes
+                .getAuthority()
+                .createOAuth2Strategy(strategyParameters);
+
+        // DCF protocol step 1: Get user code
+        // Populate global authorization request
+        mAuthorizationRequest = getAuthorizationRequest(oAuth2Strategy, parametersWithScopes);
+
+        // Call method defined in oAuth2Strategy to request authorization
+        final AuthorizationResult authorizationResult = oAuth2Strategy.getDeviceCode((MicrosoftStsAuthorizationRequest) mAuthorizationRequest, null);
+
+        validateServiceResult(authorizationResult);
+
+        Logger.verbose(
+                TAG + methodName,
+                "Device Code Flow authorization step finished..."
+        );
+        logResult(TAG, authorizationResult);
+
+        // End telemetry with LOCAL_DEVICE_CODE_FLOW_ACQUIRE_URL_AND_CODE
+        Telemetry.emit(
+                new ApiEndEvent()
+                        .putApiId(TelemetryEventStrings.Api.LOCAL_DEVICE_CODE_FLOW_ACQUIRE_URL_AND_CODE)
+        );
+
+        return authorizationResult;
     }
 
     @Override
-    public AcquireTokenResult acquireDeviceCodeFlowToken(final AuthorizationResult authorizationResult, DeviceCodeFlowCommandParameters commandParameters) throws Exception {
-        // TODO: Placeholder to avoid inheritance error. Will be implemented after Command/Controller level PR in Common
-        return null;
+    public AcquireTokenResult acquireDeviceCodeFlowToken(
+            final AuthorizationResult authorizationResult,
+            final DeviceCodeFlowCommandParameters parameters)
+            throws MsalException, ClientException, IOException, InterruptedException {
```

Comment:
Do we actually want to throw all these exceptions? Or should this method handle some of these directly?

Reply:
I think it's fine as is? MsalException should be thrown since we want to communicate that to the command class. I feel the other three would point to an error on our end.